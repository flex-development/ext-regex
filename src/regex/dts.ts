/**
 * @file Regular Expressions - EXT_DTS_REGEX
 * @module ext-regex/regex/dts
 */

/**
 * TypeScript declaration file extension regex.
 *
 * Supported extensions:
 *
 * - `.d.cts`
 * - `.d.mts`
 * - `.d.ts`
 *
 * Captured groups:
 *
 * - `type`: Letter between `'.'` and `'ts'` in file extension
 *
 * @see https://regex101.com/r/LD2eKA
 *
 * @const {RegExp} EXT_DTS_REGEX
 */
const EXT_DTS_REGEX: RegExp = /\.d\.(?<type>[cm])?ts(?=\s*$)/

export default EXT_DTS_REGEX
