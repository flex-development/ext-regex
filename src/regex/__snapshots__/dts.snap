// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`unit:EXT_DTS_REGEX > .d.cts > should match extension ".d.cts" 1`] = `
{
  "0": ".d.cts",
  "1": "c",
  "groups": {
    "type": "c",
  },
}
`;

exports[`unit:EXT_DTS_REGEX > .d.cts > should match module id given last extension is ".d.cts" 1`] = `
{
  "0": ".d.cts",
  "1": "c",
  "groups": {
    "type": "c",
  },
}
`;

exports[`unit:EXT_DTS_REGEX > .d.cts > should match module id given only extension is ".d.cts" 1`] = `
{
  "0": ".d.cts",
  "1": "c",
  "groups": {
    "type": "c",
  },
}
`;

exports[`unit:EXT_DTS_REGEX > .d.mts > should match extension ".d.mts" 1`] = `
{
  "0": ".d.mts",
  "1": "m",
  "groups": {
    "type": "m",
  },
}
`;

exports[`unit:EXT_DTS_REGEX > .d.mts > should match module id given last extension is ".d.mts" 1`] = `
{
  "0": ".d.mts",
  "1": "m",
  "groups": {
    "type": "m",
  },
}
`;

exports[`unit:EXT_DTS_REGEX > .d.mts > should match module id given only extension is ".d.mts" 1`] = `
{
  "0": ".d.mts",
  "1": "m",
  "groups": {
    "type": "m",
  },
}
`;

exports[`unit:EXT_DTS_REGEX > .d.ts > should match extension ".d.ts" 1`] = `
{
  "0": ".d.ts",
  "1": undefined,
  "groups": {
    "type": undefined,
  },
}
`;

exports[`unit:EXT_DTS_REGEX > .d.ts > should match module id given last extension is ".d.ts" 1`] = `
{
  "0": ".d.ts",
  "1": undefined,
  "groups": {
    "type": undefined,
  },
}
`;

exports[`unit:EXT_DTS_REGEX > .d.ts > should match module id given only extension is ".d.ts" 1`] = `
{
  "0": ".d.ts",
  "1": undefined,
  "groups": {
    "type": undefined,
  },
}
`;
